crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -209,6 +181,102 @@ impl Trade {
         })
     }
 
+    /// Protocol fee as a cut of surplus, denominated in SURPLUS token
+    ///
+    /// Protocol fee calculation logic depends if the protocol fee
+    /// is already applied to the trade or not. Since scoring module works with
+    /// trades that already have the protocol fee applied, we need to calculate
+    /// the protocol fee as an observation of already applied protocol fee.
```

Comment:
```suggestion
    /// Surplus fee is specified as a `factor` from raw surplus (before fee). Since this module works with
    /// trades that already have the protocol fee applied, we need to calculate
    /// the protocol fee as an observation of the eventually traded amounts using a different factor `factor'` .
```

Reply:
copied